app/helpers/application_helper.rb

```diff
@@ -127,11 +127,16 @@ def link_to_function_if_authorized(name, function, options = {}, html_options =
   end
 
   def display_delete_if_authorized(options = {}, html_options = {})
+    return "" unless authorized_for(options)
```

Comment:
Not sure why `display_delete_if_authorized` doesn't work for you here? It seems a bit risky to modify this method to me..